src/System.Text.JsonLab.Serialization/System/Text/Json/Serialization/JsonConverterSettings.cs

```diff
@@ -80,15 +81,23 @@ public int DefaultBufferSize
             }
             set
             {
-                if (value < 1)
+                if (value == 0 || value < BufferSizeUnspecified)
```

Comment:
It's odd that we allow values `<= 0`

Reply:
Why did we need to make this change? Now every call to `EffectiveBufferSize` has a branch.

Reply:
I didn't want to expose the actual constant (16K)